Source/WebCore/page/FrameView.cpp

```diff
@@ -2466,9 +2467,40 @@ void FrameView::textFragmentIndicatorTimerFired()
     TemporarySelectionChange selectionChange(document, { range }, { TemporarySelectionOption::DelegateMainFrameScroll, TemporarySelectionOption::RevealSelectionBounds, TemporarySelectionOption::UserTriggered });
     
     auto textIndicator = TextIndicator::createWithRange(range, { TextIndicatorOption::DoNotClipToVisibleRect }, WebCore::TextIndicatorPresentationTransition::Bounce);
-    if (textIndicator)
-        document.page()->chrome().client().setTextIndicator(textIndicator->data());
     
+    auto* page = frame().page();
+    
+    if (!page)
+        return;
+    
+    if (textIndicator) {
+        constexpr OptionSet<HitTestRequest::Type> hitType { HitTestRequest::Type::ReadOnly, HitTestRequest::Type::Active, HitTestRequest::Type::AllowVisibleChildFrameContentOnly };
+        
+        auto firstCharRange = resolveCharacterRange(range, { 0, 1 });
+        auto firstCharRect = RenderFlexibleBox::absoluteTextRects(firstCharRange);
+        
+        auto collapsedStart = SimpleRange(BoundaryPoint(range.startContainer(), range.startOffset()), BoundaryPoint(range.startContainer(), range.startOffset()));
+        
+        auto collapsedStartCharRects = RenderFlexibleBox::absoluteTextRects(collapsedStart);
+        
+        HitTestResult result;
+        result = page->mainFrame().eventHandler().hitTestResultAtPoint(LayoutPoint(collapsedStartCharRects.first().center()), hitType);
+        if (!intersects(range, *result.targetNode()))
+            return;
+
+        if (range.endOffset() > 1) {
```

Comment:
Odd that we can't test the end if it ends just inside a node (can that happen?).

Reply:
I think if we look for a single character word this can happen (don't know why you would, but it is valid)